### PR TITLE
Add wl_list_append()

### PR DESCRIPTION
**Pull request by johanmalm**
See #572

### CODE REVIEW

File: include/common/list.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#include <wayland-server-core.h>
+
+/**
+ * wl_list_append() - add a new element to the end of a list
+ * @list: list head to add it before
+ * @elm: new element to be added (link of the containing struct to be precise)
+ *
+ * Note: In labwc, most lists are queues where we want to add new elements to
+ * the end of the list. As wl_list_insert() adds elements at the front of the
+ * list (like a stack) - without this helper-functions - users have to add call
```
Comment (Consolatis) [inline include/common/list.h:11]: `s/add//`

nit: `s/users/we/` or `s/users have/one has/` ?

File: src/ssd/ssd_part.c [hunk 2]
```diff
@@ -41,7 +42,7 @@ add_scene_part(struct wl_list *part_list, enum ssd_part_type type)
 {
 	struct ssd_part *part = znew(*part);
 	part->type = type;
-	wl_list_insert(part_list->prev, &part->link);
+	wl_list_append(part_list, &part->link);
```
Comment (Consolatis) [inline src/ssd/ssd_part.c:45]: I like this new wrapper, so much clearer.

**Comment by Consolatis** [comment]
Do you want to get over the existing code and do a mass replace? Otherwise I could do that as well if you prefer.

**Comment by johanmalm** [comment]
> Do you want to go over the existing code and do a mass replace? Otherwise I could do that as well if you prefer.

Feel free :)
I ran out of of energy.

**Comment by Consolatis** [comment]
I think i got everything changed, also reordered some includes while at it.

File: include/common/list.h (hunk 1 above)
Comment (johanmalm) [reply include/common/list.h]: Done

**Comment by johanmalm** [comment]
I'll merge this for now.
I did a search for `wl_list_for_each_reverse` and suspect most of them should drop the reverse (and suspect the ones in `scaled_scene_buffer.c` are the exception).
I'd rather do that quite slowly and testing thoroughly thought :smile:

**Pull request merged**